File: backend/api/v1/resume_optimizer.py
from fastapi import APIRouter

from backend.schemas.resume_optimizer import (
    ResumeOptimizerRequest,
)
from backend.use_cases.optimize_resume_use_case import (
    OptimizeResumeUseCase,
)

router = APIRouter(
    prefix="/resume-optimizer",
    tags=["Resume Optimizer"],
)


@router.post("/optimize")
async def optimize_resume(
    request: ResumeOptimizerRequest,
):

    use_case = OptimizeResumeUseCase()

    return use_case.execute(
        request.resume_data,
        request.target_role,
    )

File: web/src/services/resumeOptimizer.ts
import api from "./api";

export async function optimizeResume(
  resumeData: any,
  targetRole: string
) {
  const response = await api.post(
    "/resume-optimizer/optimize",
    {
      resume_data: resumeData,
      target_role: targetRole,
    }
  );

  return response.data;
}